### PR TITLE
Add AdamW to optimizers and ReLU to fc activations

### PR DESCRIPTION
**Pull request by Multihuntr**
I think these options should be included in NeuralHydrology, for the reasons noted below:

- optimizers: Add AdamW. AdamW is effectively Adam, but with weight decay fixed (there was an mathematical error in the original formulation). In my experience in other projects, AdamW - with default weight decay - gives a small, but very consistent performance boost compared to Adam.
- modelzoo/fc: Add ReLU. ReLU is the most common activation function for Linear and Conv layers.

### CODE REVIEW

**Review by gauchm — COMMENTED** [review]
Thanks, looks good!
Nit: could you update the [docstring](https://github.com/neuralhydrology/neuralhydrology/pull/169/files#diff-4fed7cb03031130db7cabaf6d9aa8d57a95e54eb559398883a28b893727c292aR17) and [docs](https://github.com/neuralhydrology/neuralhydrology/blob/5436973d7645cf090d0d75b0ffd1d6b7f902cf68/docs/source/usage/config.rst?plain=1#L352) to reflect the new adamw option and [docs](https://github.com/neuralhydrology/neuralhydrology/blob/5436973d7645cf090d0d75b0ffd1d6b7f902cf68/docs/source/usage/config.rst?plain=1#L337) to reflect relu?

**Comment by Multihuntr** [comment]
Absolutely! I apologise that I forgot.

File: neuralhydrology/training/__init__.py [hunk 1]
```diff
@@ -30,6 +30,8 @@ def get_optimizer(model: torch.nn.Module, cfg: Config) -> torch.optim.Optimizer:
     """
     if cfg.optimizer.lower() == "adam":
         optimizer = torch.optim.Adam(model.parameters(), lr=cfg.learning_rate[0])
+    elif cfg.optimizer.lower() == "adamw":
+        optimizer = torch.optim.AdamW(model.parameters(), lr=cfg.learning_rate[0])
```
Comment (kratzert) [inline neuralhydrology/training/__init__.py:34]: Is the default weight_decay parameter good? Haven't used AdamW much but I wonder if we want to make this a configurable parameter or not. @gauchm WDYT?

File: docs/source/usage/config.rst [hunk 2]
```diff
@@ -349,7 +350,7 @@ Training settings
 -----------------
 
 -  ``optimizer``: Specify which optimizer to use. Currently supported
-   is Adam (standard). New optimizers can be added
+   are Adam (default) and AdamW. New optimizers can be added
```
Comment (kratzert) [inline docs/source/usage/config.rst:353]: The 
```suggestion
   are Adam and AdamW. New optimizers can be added
```
If I understand our `get_optimizer` function correctly, then Adam is not a default optimizer. The optimizer has to be set explicitly by the user, otherwise https://github.com/neuralhydrology/neuralhydrology/blob/master/neuralhydrology/utils/config.py#L620 will throw an error.

File: neuralhydrology/training/__init__.py (hunk 1 above)
Comment (Multihuntr) [reply neuralhydrology/training/__init__.py:34]: In my experience, the default parameter is good, and doesn't need to be configured like learning rate does. Theoretically weight decay is best for preventing overfitting, so if you think that is likely to be a problem, then I vote to make it configurable.

Comment (gauchm) [reply neuralhydrology/training/__init__.py:34]: No strong opinion either way, but for me it's fine to leave it as is for now. If we start on that, we might also want to make the betas configurable and then it already get quite a lot of new config args. It's easy to hack in for someone who really wants to try it.

Comment (kratzert) [reply neuralhydrology/training/__init__.py:34]: SGTM

**Review by gauchm — APPROVED** [review]
I think we're good to go, thanks again!

Please use "squash and merge" when merging to avoid cluttering the commit history :)

**Comment by Multihuntr** [comment]
> Please use "squash and merge" when merging to avoid cluttering the commit history :)

If that was directed at me, I don't have write access to this repo. Someone else will have to click the button.
